test/letters/utils/helpers.unit.spec.jsx

```diff
@@ -135,5 +137,90 @@ describe('Letters helpers: ', () => {
       expect(resetAddress.zipCode).to.equal('');
     });
   });
+
+  describe('stripEmpties', () => {
+    const inputAddress = {
+      addressEffectiveDate: '2012-04-03T04:00:00.000+00:00',
+      addressOne: '57 COLUMBUS STRASSA',
+      addressThree: '',
+      addressTwo: '',
+      militaryPostOfficeTypeCode: 'APO',
+      militaryStateCode: 'AE',
+      type: 'MILITARY',
+      zipCode: '09028',
+      zipSuffix: ''
+    };
+
+    const expectedAddress = {
+      addressEffectiveDate: '2012-04-03T04:00:00.000+00:00',
+      addressOne: '57 COLUMBUS STRASSA',
+      militaryPostOfficeTypeCode: 'APO',
+      militaryStateCode: 'AE',
+      type: 'MILITARY',
+      zipCode: '09028',
+    };
+
+    it('should only remove all zero-length properties from an object', () => {
+      const actualAddress = stripEmpties(inputAddress);
+      expect(actualAddress).to.eql(expectedAddress);
+    });
+  });
+
+  describe('military to generic', () => {
+    const militaryAddress = {
+      addressEffectiveDate: '2012-04-03T04:00:00.000+00:00',
+      addressOne: '57 COLUMBUS STRASSA',
+      addressThree: '',
+      addressTwo: '',
+      militaryPostOfficeTypeCode: 'APO',
+      militaryStateCode: 'AE',
+      type: 'MILITARY',
+      zipCode: '09028',
+      zipSuffix: ''
+    };
+
+    const domesticAddress = {
+      addressEffectiveDate: '2012-04-03T04:00:00.000+00:00',
+      addressOne: '57 COLUMBUS STRASSA',
+      addressThree: '',
+      addressTwo: '',
+      city: 'Chicago',
+      stateCode: 'IL',
+      countryName: 'USA',
+      type: 'DOMESTIC',
+      zipCode: '06628',
+      zipSuffix: ''
+    };
+
+    const genericAddress = {
+      addressEffectiveDate: '2012-04-03T04:00:00.000+00:00',
+      addressOne: '57 COLUMBUS STRASSA',
+      addressThree: '',
+      addressTwo: '',
+      city: 'APO',
+      stateCode: 'AE',
+      countryName: 'USA',
+      type: 'MILITARY',
+      zipCode: '09028',
+      zipSuffix: ''
+    };
+
+    it('translates military addresses to generic', () => {
+      const actualAddress = militaryToGeneric(militaryAddress);
+      expect(actualAddress).to.eql(genericAddress);
+    });
+
+    it('lets non-military address types pass through unmodified', () => {
+      const actualAddress = militaryToGeneric(domesticAddress);
+      expect(actualAddress).to.eql(domesticAddress);
+    });
+
+    it('returns a clone for all cases', () => {
+      const militaryTest = militaryToGeneric(militaryAddress);
+      const nonMilitaryTest = militaryToGeneric(domesticAddress);
+      expect(militaryTest).to.not.equal(militaryAddress);
+      expect(nonMilitaryTest).to.not.equal(domesticAddress);
```

Comment:
See, this is tricky because the "top-level" objects _are_ different refs, but the "sub" objects will have the same refs.

For the address objects, though, it's probably totally fine because we won't have any nested objects. Suppose this is more academic then anything, really. 🤔

Reply:
Yeah, so these are just shallow objects with string values. But, that's just an implicit invariant and not actually specced out anywhere in the codebase, which is probably not great.